### PR TITLE
Synch tree counter

### PR DESCRIPTION
**Pull request by vauxhall**
*Thank you for submitting a pull request, your contributions are greatly appreciated!*

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Please make sure you've run the test scheme and all the tests pass. If your changes affect existing tests please make sure to update the tests as well.

### CODE REVIEW

File: Client/Frontend/Browser/Tab.swift [hunk 1]
```diff
@@ -435,27 +435,29 @@ class Tab: NSObject {
...
             if let url = request.url, url.isFileURL, request.isPrivileged {
-                return webView.loadFileURL(url, allowingReadAccessTo: url)
+                completion?(webView.loadFileURL(url, allowingReadAccessTo: url))
             }
 
             // Ecosia: inject cookie
-            var request = request
             if let cookie = browserViewController?.tabManager.cookie {
+                var request = request
                 request.allHTTPHeaderFields = HTTPCookie.requestHeaderFields(with: [cookie.value])
```
Comment (ecotopian) [inline Client/Frontend/Browser/Tab.swift:455]: so as I see it injecting the cookie via `request.allHTTPHeaderFields ` will make it work for private mode but I think we need to use `webView.configuration.websiteDataStore.httpCookieStore.setCookie` for non private tabs to keep logged in state for new tabs. If not, we would not need the PR at all as it apparently does not change the way the cookie is injected or am I overseeing anything?

**Comment by vauxhall** [comment]
@ecotopian Ready for another review 👍

File: Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -6,7 +6,7 @@
         "repositoryURL": "https://github.com/ecosia/ios-core.git",
         "state": {
           "branch": "main",
-          "revision": "6098b2de91d43917743ed1883eced133d07108f2",
+          "revision": "7ef7ba234816b909f801574389a72c32f88fd7d4",
```
Comment (ecotopian) [inline Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:9]: I suggest that for the next release we create another tag for core and hardwire this one.. but not needed now 👍

**Review by ecotopian — APPROVED** [review]
Looks great! 👍

File: Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 2 above)
Comment (vauxhall) [reply Client.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:9]: The core release is already in place but as this was pointed to main I opted not to change it.

**Pull request merged**